components/dashboard/src/start/CreateWorkspace.tsx

```diff
@@ -150,5 +150,60 @@ export class CreateWorkspace extends React.Component<CreateWorkspaceProps, Creat
       </div>}
     </StartPage>;
   }
+}
+
+interface RunningPrebuildViewProps {
+  runningPrebuild: {
+    prebuildID: string
+    workspaceID: string
+    starting: RunningWorkspacePrebuildStarting
+    sameCluster: boolean
+  };
+  onIgnorePrebuild: () => void;
+  onPrebuildSucceeded: () => void;
+}
+
+function RunningPrebuildView(props: RunningPrebuildViewProps) {
+  const logsEmitter = new EventEmitter();
+  const service = getGitpodService();
+  let pollTimeout: NodeJS.Timeout | undefined;
+
+  useEffect(() => {
+    const pollIsPrebuildDone = async () => {
+      clearTimeout(pollTimeout!);
+      const available = await service.server.isPrebuildDone(props.runningPrebuild.prebuildID);
+      if (available) {
+        props.onPrebuildSucceeded();
+        return;
+      }
+      pollTimeout = setTimeout(pollIsPrebuildDone, 10000);
+    };
+    const watchPrebuild = () => {
+      service.server.watchHeadlessWorkspaceLogs(props.runningPrebuild.workspaceID);
+      pollIsPrebuildDone();
+    };
+    watchPrebuild();
+
+    const toDispose = service.registerClient({
+      notifyDidOpenConnection: () => watchPrebuild(),
+      onHeadlessWorkspaceLogs: event => {
+        if (event.workspaceID !== props.runningPrebuild.workspaceID) {
+          return;
+        }
+        logsEmitter.emit('logs', event.text);
+      },
+    });
 
+    return function cleanup() {
+      clearTimeout(pollTimeout!);
+      toDispose.dispose();
+    };
+  });
```

Comment:
Looks like this should only run on `componentDidMount` and not every update.
Therefore, you should pass an empty dependency array as the second argument. See https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

Reply:
That said, in practice there seem to be no `componentDidUpdate` calls in this case. But still it would be good to change it so that readers don't get bothered.

Reply:
Good point, thanks! Today I learned about `useEffect` dependencies.

Added empty dependency array to all 3 similar components.